com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/TextureGenerator.java

```diff
@@ -177,8 +187,53 @@ else if (componentCount == 3)
         return targetFormat;
     }
 
+    private static ByteBuffer getByteBuffer(BufferedImage bi)
+    {
+        ByteBuffer byteBuffer;
+        DataBuffer dataBuffer = bi.getRaster().getDataBuffer();
+
+        if (dataBuffer instanceof DataBufferByte) { // This is the usual case, where data is simply wrapped
+            byte[] pixelData = ((DataBufferByte) dataBuffer).getData();
+            byteBuffer = ByteBuffer.wrap(pixelData);
+        }
```

Comment:
Huge gain, we simply wrap the image data, as it's already in byte layout.
(Although in ABGR)

Reply:
isn't it better to prepare texture in bob and that use just memcpy in runtime? I mean we always try to make everything we can during the build process

Reply:
Not sure what you  mean?
This _is_ the build process?
We pass the data to the c++ code (texc.cpp) for processing.
We don't do the compression etc in Java.